lib/active_merchant/billing/gateways/redsys_rest.rb

```diff
@@ -507,6 +519,21 @@ def encrypt(key, order_id)
       def mac256(key, data)
         OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), key, data)
       end
+
+      def filter_merchant_parameters(transcript)
+        # Enhancement, the gateway response with base64 and it contians sensible data.
+        # Decode + Scrub + Encode the returned sensitive dat.
+        pre_filter_data =  transcript.match(%r(Ds_MerchantParameters=(\w+)))
+        decoded_pre_filter_data = Base64.decode64(pre_filter_data[1])
```

Comment:
For me, the `test_scrub` is failing here:

```
Error: test_scrub(RedsysRestTest):
  NoMethodError: undefined method `[]' for nil:NilClass
  
          decoded_pre_filter_data = Base64.decode64(pre_filter_data[1])
```

Reply:
Thanks, a return if no pre_filter_data is nil could work.